### PR TITLE
Feature 26 editin page properties

### LINKED ISSUE #26: Editing page properties
As a user I'd like to be able to edit page properties.

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #26

### CODE REVIEW

File: src/components/editor/page-properties.tsx [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
-import { LayoutType, QuestionType } from "../../types";
 import GenericDialog from "../generic/generic-dialog";
 import WithDebounce from "../generic/with-debounce";
-import { Close, Edit } from "@mui/icons-material";
-import AddCircleIcon from "@mui/icons-material/AddCircle";
+import { AddCircle, Edit } from "@mui/icons-material";
 import {
   Box,
   Button,
-  IconButton,
```
Comment (EVP1989) [inline src/components/editor/page-properties.tsx:15]: Comment to test github, resolve.

**Review by EVP1989 — APPROVED** [review]

File: src/component-renderer/title-renderer.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { v4 as uuid } from "uuid";
+
+/**
+ * Class for PageTitleRenderer
+ */
+export class PageTitleRenderer {
+  /**
+   * Convert single question type options into html buttons
```
[the remark this answers is not in the document]
Comment (villejuutila) [reply src/component-renderer/title-renderer.ts:8]: Resolved

File: src/component-renderer/button-renderer.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { v4 as uuid } from "uuid";
+
+/**
+ * Class for ButtonRenderer
+ */
+export class ButtonRenderer {
```
[the remark this answers is not in the document]
Comment (villejuutila) [reply src/component-renderer/button-renderer.ts:6]: Resolved

File: src/components/editor/page-properties.tsx [hunk 4]
```diff
@@ -33,275 +38,288 @@ interface Props {
...
+      question: {
+        ...pageToEdit.question,
+        options: [
+          ...pageToEdit.question.options.map((opt) =>
+            option === opt ? { ...option, questionOptionValue: value } : opt
+          )
+        ]
+      }
+    };
+    console.log(pageToUpdate);
```
[the remark this answers is not in the document]
Comment (villejuutila) [reply src/components/editor/page-properties.tsx]: Resolved

File: src/utils/page-utils.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  type: PageElementType;
+  element: Element;
+  id: string;
+};
+
+export enum PageElementType {
+  H1 = "h1",
+  P = "p",
+  DIV = "div"
+}
```
[the remark this answers is not in the document]
Comment (villejuutila) [reply src/utils/page-utils.ts:121]: Resolved

File: src/utils/preview-utils.ts [hunk 6]
```diff
@@ -1,8 +1,39 @@
...
+  html: string,
+  layoutVariables: LayoutVariable[],
+  pageProperties: PageProperty[]
+) => {
+  const body = new DOMParser().parseFromString(html, "text/html").body;
+  return body;
+};
+
+/**
+ * TODO: ADD DOCS
```
[the remark this answers is not in the document]
Comment (villejuutila) [reply src/utils/preview-utils.ts:23]: Removed whole method as unused therefore resolved

File: src/utils/preview-utils.ts [hunk 7]
```diff
@@ -1,8 +1,39 @@
...
+ */
+const updateHtmlElement = (element: Element, pageProperties: PageProperty[]) => {
+  const elementId = element.id;
+  const pageProperty = pageProperties.find((property) => property.key === elementId);
+  console.log(pageProperty);
```
[the remark this answers is not in the document]
Comment (villejuutila) [reply src/utils/preview-utils.ts:28]: Resolved.

**Pull request merged**